include/deal.II/base/quadrature_lib.h

```diff
@@ -438,13 +438,13 @@ class QSorted : public Quadrature<dim>
   /**
    * The constructor takes an arbitrary quadrature formula.
    */
-  QSorted (const Quadrature<dim>);
+  QSorted (const Quadrature<dim> &quad);
 
   /**
    * A rule to reorder pairs of points and weights.
    */
-  bool operator()(const std::pair<double, Point<dim> > &a,
-                  const std::pair<double, Point<dim> > &b);
+  bool operator()(const std::size_t &a,
+                  const std::size_t &b) const;
```

Comment:
I introduced a new private function `compare_weights` instead. What should happen to the operator? I doubt anybody used it (nothing you can externally use it for really), so should I just remove it? Or deprecate it first?